### DISCUSSION TITLE
About AutoML Time Series Model Fitting, Saving and Loading

### DISCUSSION THREAD

**Discussion by mertcan-yukselsin** [pycaret/pycaret#1426] [Q&A] [answered]
Hello, I have 2 questions about newly developed AutoML flow for Time Series using the new PyCaret Time Series Module (Alpha Release).

1) While fitting AutoML Time series models, we only include the data itself and it automatically splits the data into train and test datasets. Then, we can obtain test set predictions and unknown future. Does the system includes the test data to the model while fitting in order to forecast the future? Or does it automatically fit models (only to the train set) and obtain a final result to forecast test+unknown_future together ? If it's the latter, can we directly choose the model not splitting test set?

2)  In the 101 tutorial, it was seen that in the end, we can save the model and use it after loading the model. However, I couldn't see any part that how the model itself can be used to forecast future when a new observation comes. In other words, can we use the final tuned model when a new dataset comes to the picture (for example we have fitted the model using 1200 observations (monthly data) and for the next month, a new observation comes and now I want to fit the same final model including that last point and forecast again)?

Thank you!

**Comment by ngupta23** [comment, marked as answer]
@mertcan-yukselsin Sorry, I somehow missed this discussion post earlier.

* Regarding the alpha release, AutoML is not implemented yet, but you can create your own flow. Example

```python
best_model = compare_models()  # only trains using the train set
predict_model(best_model ) # predicts the test set

tuned_best_model = tune_model(best_model)  # performs hyperparameter tuning using train data only
predict_model(tuned_best_model) # predicts the test set

final_model = finalize_model(tuned_best_model) # uses best hyperparameters and trains on entire dataset
predict_model(final_model) # predicts true future data (actuals unknown)
```

* Regarding incremental (online) learning when a new monthly data comes in...

This is a good suggestion. It has not been implemented yet but would be a good addition. You will have to simply retrain the model with the entire dataset for now.

* **Also, you may be interested in seeing the time series roadmap. You can find it [here](https://github.com/pycaret/pycaret/issues/1648)**.

**Comment by ngupta23** [comment]
>In the 101 tutorial, it was seen that in the end, we can save the model and use it after loading the model. However, I couldn't see any part that how the model itself can be used to forecast future when a new observation comes. In other words, can we use the final tuned model when a new dataset comes to the picture (for example we have fitted the model using 1200 observations (monthly data) and for the next month, a new observation comes and now I want to fit the same final model including that last point and forecast again)?

@mertcan-yukselsin I have added a FAQ on how you can do online learning using `pycaret` and `sktime` now. I hope that answers your question

https://github.com/pycaret/pycaret/discussions/1767
